core/src/main/java/org/frankframework/validation/SchemaUtils.java

```diff
@@ -117,67 +120,96 @@ public static void mergeRootXsdsGroupedByNamespaceToSchemasWithIncludes(Map<Stri
-				resultXsd.setSchemaContentsWriter(schemaContentsWriter);
-				w = XmlUtils.REPAIR_NAMESPACES_OUTPUT_FACTORY.createXMLStreamWriter(schemaContentsWriter);
-			} else {
-				w = xmlStreamWriter;
-			}
-			int i = 0;
-			// perform the merge
-			for (IXSD xsd: xsds) {
-				i++;
-				boolean skipRootElementStart;
-				boolean skipRootElementEnd;
-				if (xsds.size() == 1) {
-					skipRootElementStart = false;
-					skipRootElementEnd = false;
-				} else {
-					skipRootElementStart = i != 1;
-					skipRootElementEnd = i < xsds.size();
-				}
-				xsdToXmlStreamWriter(xsd, w, false, true, skipRootElementStart, skipRootElementEnd, rootAttributes, rootNamespaceAttributes, imports, false);
-			}
-			// TODO: After creating merged XSD, while we still know what the source files are, we should now validate for duplicate elements
-			if (resultXsd != null) {
-				IXSD firstXsd = xsds.iterator().next();
-				resultXsd.setImportedSchemaLocationsToIgnore(firstXsd.getImportedSchemaLocationsToIgnore());
-				resultXsd.setUseBaseImportedSchemaLocationsToIgnore(firstXsd.isUseBaseImportedSchemaLocationsToIgnore());
-				resultXsd.setImportedNamespacesToIgnore(firstXsd.getImportedNamespacesToIgnore());
-				resultXsd.initFromXsds(namespace, scopeProvider, xsds);
-				resultXsds.add(resultXsd);
-			}
+			mergeXsds(scopeProvider, xsds, namespace, xmlStreamWriter);
+
+			xmlStreamWriter.close();
+			StringXsd resultXsd = new StringXsd(contents.toString());
+			IXSD firstXsd = xsds.iterator().next();
+			resultXsd.setImportedSchemaLocationsToIgnore(firstXsd.getImportedSchemaLocationsToIgnore());
+			resultXsd.setUseBaseImportedSchemaLocationsToIgnore(firstXsd.isUseBaseImportedSchemaLocationsToIgnore());
+			resultXsd.setImportedNamespacesToIgnore(firstXsd.getImportedNamespacesToIgnore());
+			resultXsd.initFromXsds(namespace, scopeProvider, xsds);
+
+			resultXsds.add(resultXsd);
 		}
 		return resultXsds;
 	}
 
+	private static void mergeXsds(IScopeProvider scopeProvider, Set<IXSD> xsds, String namespace, XMLStreamWriter xmlStreamWriter) throws IOException, ConfigurationException {
+		if (xsds.isEmpty()) {
+			return;
+		}
+		addXsdMergeWarnings(scopeProvider, xsds, namespace);
+
+		// Get attributes of root elements and get import elements from all XSDs
+		List<Attribute> rootAttributes = new ArrayList<>();
+		List<Namespace> rootNamespaceAttributes = new ArrayList<>();
+		Set<XsdStartElementWrapper> imports = new LinkedHashSet<>();
+		String xsdPrefix = null;
+		IXSD firstXsd = null;
+		for (IXSD xsd: xsds) {
+			if (firstXsd == null) {
+				firstXsd = xsd;
+			}
+			xsdPrefix = collectImportsAndSchemaRootAttributes(xsd, rootAttributes, rootNamespaceAttributes, imports);
+		}
+		if (firstXsd == null || xsdPrefix == null) {
+			throw new IllegalStateException("Must pass at least one xsd, should have been checked already");
+		}
+		// TODO: After collecting all imports, while we still know what the source files are, we should now validate for duplicate elements
```

Comment:
todo in code - is hier een ticket voor?

Reply:
Nee, volgens mij is daar geen ticket voor. De TODO is al ouder maar ik heb hem meegekopieerd naar de nieuwe locatie.